commit c357d5fbe5d569c96aca26e8ca9467a7d69ea67a
Author: Dan Williams <dcbw@redhat.com>
Date:   Fri Jul 20 14:47:41 2012 -0500

    bluetooth: suppress another Bluez warning (rh #840580)
    
    When Bluez can't be launched for some reason, like because it
    doesn't exist, ignore that.

commit f03b0ac17681a45764c504ebdb2d96fbd7c2d227
Author: Dan Williams <dcbw@redhat.com>
Date:   Mon Dec 12 14:06:22 2011 -0600

    bluetooth: ignore "No such adapter" warnings when bluetooth is inactive

diff -up NetworkManager-0.8.1/src/bluez-manager/nm-bluez-manager.c.foo NetworkManager-0.8.1/src/bluez-manager/nm-bluez-manager.c
--- NetworkManager-0.8.1/src/bluez-manager/nm-bluez-manager.c.foo	2012-07-20 12:40:04.293947404 -0500
+++ NetworkManager-0.8.1/src/bluez-manager/nm-bluez-manager.c	2012-07-20 12:43:56.776701253 -0500
@@ -181,8 +181,12 @@ default_adapter_cb (DBusGProxy *proxy, D
 	if (!dbus_g_proxy_end_call (proxy, call, &err,
 	                            DBUS_TYPE_G_OBJECT_PATH, &default_adapter,
 	                            G_TYPE_INVALID)) {
-		nm_log_warn (LOGD_BT, "bluez error getting default adapter: %s",
-		             err && err->message ? err->message : "(unknown)");
+		/* Ignore "No such adapter" errors; just means bluetooth isn't active */
+		if (   !dbus_g_error_has_name (err, "org.bluez.Error.NoSuchAdapter")
+		    && !g_error_matches (err, DBUS_GERROR, DBUS_GERROR_SERVICE_UNKNOWN)) {
+			nm_log_warn (LOGD_BT, "bluez error getting default adapter: %s",
+			             err && err->message ? err->message : "(unknown)");
+		}
 		g_error_free (err);
 		return;
 	}
